lsp-mode.el

```diff
@@ -2937,15 +2937,20 @@ If NO-WAIT is non-nil send the request as notification."
            resp-result resp-error done?)
       (unwind-protect
           (progn
-            (lsp-request-async method params (lambda (res) (setf resp-result (or res :finished)))
-                               :error-handler (lambda (err) (setf resp-error err))
+            (lsp-request-async method params (lambda (res)
+                                               (setf resp-result (or res :finished))
+                                               (throw 'lsp-done '_))
+                               :error-handler (lambda (err)
+                                                (setf resp-error err)
+                                                (throw 'lsp-done '_))
                                :no-merge no-merge
                                :mode 'detached
                                :cancel-token :sync-request)
-            (while (not (or resp-error resp-result))
-              (accept-process-output nil 0.001)
-              (when (< expected-time (time-to-seconds (current-time)))
-                (error "Timeout while waiting for response. Method: %s." method)))
+            (catch 'lsp-done
+              (while (not (or resp-error resp-result))
+                (sit-for 1)
+                (when (< expected-time (time-to-seconds (current-time)))
+                  (error "Timeout while waiting for response.  Method: %s" method))))
```

Comment:
I think that this should be changed to accept-process-output with timeout = lsp-response-timeout

Reply:
I think the `accept-procecss-output` will actually block input and we don't want that, do we?

Reply:
Try `(sit-for lsp-response-timeout)`.

Reply:
lsp-request should block.

Reply:
ok, got it.